src/dipdup/index.py

```diff
@@ -254,7 +256,15 @@ async def _on_match(
                     continue
 
                 parameter_type = pattern_config.parameter_type_cls
-                parameter = parameter_type.parse_obj(operation.parameter_json) if parameter_type else None
+                try:
+                    parameter = parameter_type.parse_obj(operation.parameter_json) if parameter_type else None
+                except ValidationError:
+                    error_context = dict(
+                        hash=operation.hash,
+                        counter=operation.counter,
+                        nonce=operation.nonce
+                    )
+                    raise InvalidDataError(operation.parameter_json, parameter_type, error_context)
```

Comment:
`except Error as e raise Error from e` as first exception is a direct cause of second one.